### ISSUE TITLE
Sybil Attacker Report

### ISSUE THREAD

**Issue by GreyPK** [hop-protocol/hop-airdrop#116]
# Related Addresses
20 related addresses from 1 transaction.

```
0x074b0c4fa82ea41552bdf377db9c994033a92618
0x081308124e9837a77af818ea9c53ad710ad649cf
0x0ac57793a23b4079bc1fb356fdc9716b9359fb59
0x1a0381bcac439651317a3280cd9a6457bf905a58
0x216e1925df4a4a6f55f5963a9128298da1a86f5e
0x2f1750f703d5e17104eebc7072fb145fc88a9d20
0x3323da6ae50a2f53c2b8bf85267fd695b7fb2730
0x3772774f6466c8cfb1845cca1a4586c984d9fbcd
0x3abe3868a2a773f88effb7b15f5a0b5ddf3bd662
0x59ef71d8e8e4c5e0d2108f08089767d94bdbd3eb
0x5fde0f8118d172c717cd235e340a8e1c1d45edfe
0x74c38f3eb7bb28ff3295c14a7035ad645741fe14
0x94855ca59050d2efa1833a80784c583d70ab20fd
0x9bdefab54f484500e5f842e37202125aa4e92ad1
0xa5a1164ac9013a26bce393588268486959e996d9
0xb0758834ca0e6864bf5211684dd1eee386443bb6
0xbcdf8c5b369a2b94dfae69251a1349c3de3071e6
0xcfe0770f9e52607d4bb5080329c8035143b3a318
0xec2c038110cabb4d921740af5de16f84e70b9bcf
0xf58ecb469683ce087eebb56aa8183bd417e30f60
```

# Reasoning

1) Address 0xa6b0877770d20babcdbdde683dcf5a29ded511bc used Disperse.app to send MATIC to these addresses in 1 transaction. 20 of 100 recipients addresses of that thx send is eligible for the airdrop. This transaction: https://polygonscan.com/tx/0x96e7c873565efa8fc2b273be051e1cb64e0ab758017068edd0e2beadfd6432aa

2) All these 20 wallets used HOP swapAndSend in almost same day (6-8 Dec 2021) with always 110 USDC.

0x074b0c4fa82ea41552bdf377db9c994033a92618 Dec-08-2021 05:08:06 PM +UTC https://polygonscan.com/tx/0xab9687e759bfe40b0e5e02e369a377b072d9f8e0036872d403cba6854735d521
0x081308124e9837a77af818ea9c53ad710ad649cf Dec-08-2021 02:09:13 PM +UTC https://polygonscan.com/tx/0x64413cd7d0485081145287d2cf0a90138b19c6d3049a41ee224f853539372087
0x0ac57793a23b4079bc1fb356fdc9716b9359fb59 Dec-08-2021 03:42:12 PM +UTC https://polygonscan.com/tx/0x801e6785db87e4076092b847218995d60042d65181fcf90a47fa96a162805680
0x1a0381bcac439651317a3280cd9a6457bf905a58 Dec-08-2021 04:46:32 PM +UTC https://polygonscan.com/tx/0xdc141ba39c79eec4e94049ed91ad912e31377d82c9d516f65a73a360f015180c
0x216e1925df4a4a6f55f5963a9128298da1a86f5e Dec-06-2021 09:24:11 AM +UTC https://polygonscan.com/tx/0x96452f346387cf293782af3a05635aa082b30f6070dfe443194ede5378026244
0x2f1750f703d5e17104eebc7072fb145fc88a9d20 Dec-08-2021 07:24:30 AM +UTC https://polygonscan.com/tx/0xebac7b4fdb5d9a6467042a94f54f7804c3d7897a48b0d861fe5cc552ab954555
0x3323da6ae50a2f53c2b8bf85267fd695b7fb2730 Dec-08-2021 05:16:48 PM +UTC https://polygonscan.com/tx/0x8bc4c141f69b87d67dfc3611a66894e6f4a85e010394440e96e5bfb9dc181eb5
0x3772774f6466c8cfb1845cca1a4586c984d9fbcd Dec-08-2021 07:33:22 AM +UTC https://polygonscan.com/tx/0x9444a49e71c14ec8783836d04f29a2d16fec80f8b51a70331c86e270d57531ae
0x3abe3868a2a773f88effb7b15f5a0b5ddf3bd662 Dec-08-2021 02:50:50 PM +UTC https://polygonscan.com/tx/0x0acc3f87409a15acc6999cbd57db46c68258d545f344b4615586e78e837bde21
0x59ef71d8e8e4c5e0d2108f08089767d94bdbd3eb Dec-08-2021 03:11:00 PM +UTC https://polygonscan.com/tx/0x1e70ac4902a60d2220d141f278ab7c64cc6c254fa4fe4e67b0ceb4578efd0453
0x5fde0f8118d172c717cd235e340a8e1c1d45edfe Dec-07-2021 05:32:47 PM +UTC https://polygonscan.com/tx/0x8d4fadc3c1922595be16c65c69f955596fc347c1bfb35b60a1435849e68fcd5f
0x74c38f3eb7bb28ff3295c14a7035ad645741fe14 Dec-08-2021 01:49:57 PM +UTC https://polygonscan.com/tx/0x3755210c70793d2cf3836e8f3d122447b9a44023a1d4d54363b21eb23a4b8b83
0x94855ca59050d2efa1833a80784c583d70ab20fd Dec-08-2021 04:27:50 PM +UTC https://polygonscan.com/tx/0x91a48da92276268634ceab530a55be1d01b13bd6e37e27764e6305cdfcdc32db
0x9bdefab54f484500e5f842e37202125aa4e92ad1 Dec-08-2021 02:15:47 PM +UTC https://polygonscan.com/tx/0x99aaff8f25cf0d44fdc292bf418c193690d6b5d7425b0f4ab880db1c62257cd5
0xa5a1164ac9013a26bce393588268486959e996d9 Dec-07-2021 04:27:10 PM +UTC https://polygonscan.com/tx/0x3ada469dd2805a5cfd8bd0a38c3317104d033720d71b072ef51d64ddfc1bfffd
0xb0758834ca0e6864bf5211684dd1eee386443bb6 Dec-07-2021 04:49:46 PM +UTC https://polygonscan.com/tx/0x36f377f721f9f0fa31e0807c0b063c90fe55eb7d07a17310d614c935be5a319d
0xbcdf8c5b369a2b94dfae69251a1349c3de3071e6 Dec-08-2021 12:53:09 PM +UTC https://polygonscan.com/tx/0x780a5269d95010185dffbbc9134c4cc98ae713bc6002cec520054d3e00bf2f00
0xcfe0770f9e52607d4bb5080329c8035143b3a318 Dec-08-2021 04:59:12 PM +UTC https://polygonscan.com/tx/0x83a8bc2fd9cc4d0faca38faaeaeb8e88bf020a4623832a5e6198b37944896969
0xec2c038110cabb4d921740af5de16f84e70b9bcf Dec-08-2021 05:58:45 PM +UTC https://polygonscan.com/tx/0x59dddb3be62333d02f4265e30423af45ef413a3ea468065f9d28a749fa8aee60
0xf58ecb469683ce087eebb56aa8183bd417e30f60 Dec-08-2021 05:37:27 PM +UTC https://polygonscan.com/tx/0xc68ddcf30df4173acb77efe35db3e53a976423c4b6ec58550a991c3d353f45d6

# Methodology

I checked Disperse polygon contract's transactions with polygonscan API and filtered transactions, where 20+ recipients addresses are eligible for the airdrop. Then I checked when and how this wallets interact with HOP. Node js code  available on request.

# Rewards Address

0x7d3f6048091aEdE6198736b8C6ac527c314e25ff

**Comment by shanefontaine** [comment]
Thank you for your report @GreyPK. We have verified that the addresses in this report are Sybil attackers.

The report included 20 eligible addresses as Sybil attackers which means you are eligible for 2379.101113859814797638 HOP!

The qualified addresses are as follows:

```
0x074b0c4fa82ea41552bdf377db9c994033a92618
0x081308124e9837a77af818ea9c53ad710ad649cf
0x0ac57793a23b4079bc1fb356fdc9716b9359fb59
0x1a0381bcac439651317a3280cd9a6457bf905a58
0x216e1925df4a4a6f55f5963a9128298da1a86f5e
0x2f1750f703d5e17104eebc7072fb145fc88a9d20
0x3323da6ae50a2f53c2b8bf85267fd695b7fb2730
0x3772774f6466c8cfb1845cca1a4586c984d9fbcd
0x3abe3868a2a773f88effb7b15f5a0b5ddf3bd662
0x59ef71d8e8e4c5e0d2108f08089767d94bdbd3eb
0x5fde0f8118d172c717cd235e340a8e1c1d45edfe
0x74c38f3eb7bb28ff3295c14a7035ad645741fe14
0x94855ca59050d2efa1833a80784c583d70ab20fd
0x9bdefab54f484500e5f842e37202125aa4e92ad1
0xa5a1164ac9013a26bce393588268486959e996d9
0xb0758834ca0e6864bf5211684dd1eee386443bb6
0xbcdf8c5b369a2b94dfae69251a1349c3de3071e6
0xcfe0770f9e52607d4bb5080329c8035143b3a318
0xec2c038110cabb4d921740af5de16f84e70b9bcf
0xf58ecb469683ce087eebb56aa8183bd417e30f60
```

**Comment by shanefontaine** [comment]
In addition to the above similarities, all addresses:

* Had activity on 8 or 9 networks
* Had < 0.1 MATIC on Polygon at the time of review
* Exhibited near identical behavior on Hop protocol

address | ensName | total | mainnet | arbitrum | optimism | polygon | xdai | totalVolume
-- | -- | -- | -- | -- | -- | -- | -- | --
0xf58ecb469683ce087eebb56aa8183bd417e30f60 |   | 19 | 0 | 0 | 0 | 10 | 9 | 2198.665191
0x5fde0f8118d172c717cd235e340a8e1c1d45edfe |   | 19 | 0 | 0 | 0 | 10 | 9 | 2198.137027
0x59ef71d8e8e4c5e0d2108f08089767d94bdbd3eb |   | 19 | 0 | 0 | 0 | 9 | 10 | 1979.396343
0x3abe3868a2a773f88effb7b15f5a0b5ddf3bd662 |   | 18 | 0 | 0 | 0 | 9 | 9 | 1979.096596
0x2f1750f703d5e17104eebc7072fb145fc88a9d20 |   | 18 | 0 | 0 | 0 | 9 | 9 | 1978.96484
0x3772774f6466c8cfb1845cca1a4586c984d9fbcd |   | 18 | 0 | 0 | 0 | 9 | 9 | 1978.658436
0xb0758834ca0e6864bf5211684dd1eee386443bb6 |   | 18 | 0 | 0 | 0 | 9 | 9 | 1978.313521
0x216e1925df4a4a6f55f5963a9128298da1a86f5e |   | 18 | 0 | 0 | 0 | 9 | 9 | 1977.990145
0x9bdefab54f484500e5f842e37202125aa4e92ad1 |   | 18 | 0 | 0 | 0 | 9 | 9 | 1977.98778
0x1a0381bcac439651317a3280cd9a6457bf905a58 |   | 18 | 0 | 0 | 0 | 9 | 9 | 1977.89428
0x94855ca59050d2efa1833a80784c583d70ab20fd |   | 18 | 0 | 0 | 0 | 9 | 9 | 1977.764562
0x081308124e9837a77af818ea9c53ad710ad649cf |   | 18 | 0 | 0 | 0 | 9 | 9 | 1977.274177
0x74c38f3eb7bb28ff3295c14a7035ad645741fe14 |   | 18 | 0 | 0 | 0 | 9 | 9 | 1977.206924
0xec2c038110cabb4d921740af5de16f84e70b9bcf |   | 18 | 0 | 0 | 0 | 9 | 9 | 1977.199539
0xbcdf8c5b369a2b94dfae69251a1349c3de3071e6 |   | 18 | 0 | 0 | 0 | 9 | 9 | 1976.944569
0x0ac57793a23b4079bc1fb356fdc9716b9359fb59 |   | 17 | 0 | 0 | 0 | 9 | 8 | 1868.881454
0x074b0c4fa82ea41552bdf377db9c994033a92618 |   | 16 | 0 | 0 | 0 | 8 | 8 | 1758.894383
0xa5a1164ac9013a26bce393588268486959e996d9 |   | 16 | 0 | 0 | 0 | 8 | 8 | 1758.33692
0x3323da6ae50a2f53c2b8bf85267fd695b7fb2730 |   | 16 | 0 | 0 | 0 | 8 | 8 | 1758.267882
0xcfe0770f9e52607d4bb5080329c8035143b3a318 |   | 16 | 0 | 0 | 0 | 8 | 8 | 1758.120469

**Comment by forweb3** [comment]
sir ，One of my, our report is finding that the rule has to be a jump to survive. However, the successor has completed more than 10 user tracking records, and there is no record of him. I cancelled his reward. And this is what I found so am I not rewarded accordingly? This address has not interacted with hop. And he did not provide other information that he interacted. I report all his reports.  
0x7d3f6048091aEdE6198736b8C6ac527c314e25ff This is a brand new empty address, why can't he provide the address he interacts with, because he is the real empty hunter. He circumvented it with better technical isolation. So here's my reasoning, I want 25% of his reward to go to me and then the remaining 75% to all eligible real users. Just like our rules.
